data/publiccloud/restart_instance.sh

```diff
@@ -50,6 +51,10 @@ case $PROVIDER in
     AZURE)
         az vm restart -g $INSTANCE_ID -n $INSTANCE_ID --no-wait
         ;;
+    GCE)
+        gcloud compute instances stop $INSTANCE_ID --zone $ZONE --async
```

Comment:
stop & start is the same as doing reset command:
https://cloud.google.com/compute/docs/instances/stop-start-instance#resetting_an_instance

```suggestion
        gcloud compute instances reset $INSTANCE_ID --zone $ZONE --async
```

Reply:
yeah I have seen this , but I got afraid by this : 
```
Caution: Resetting an instance is a disruptive action; the VM does not do a graceful shutdown of the guest OS and all application state and memory is lost. Make sure you understand the implications of this before resetting your VM.
```

while `stop` is graceful shutdown . As far as I understand we doing restart to keep using VM , right ? So we do care if it would be able to actually start afterwards , right ? Not saying that it **won't** start after reset but I think there is less chances , right ? :)

Reply:
I am not sure stopping is a graceful stop.. Looking at the description [here](https://cloud.google.com/compute/docs/instances/stop-start-inst) looks like it looses _its in-memory data and virtual machine state_ which sound similar to reset  _Resetting an instance forcibly wipes the memory contents of the machine and resets the virtual machine to its initial state._